File: funciones.js
/* Funciones */
/* 

1. Funciones de Nombre:

function nombreFuncion(params){
    codigo
}

2. Funciones Declarativas

const nombreFuncion = function(params){
    codigo
}

3. Funcion Flecha

const nombreFuncion = (params) => codigo 

*/

function sumar(a, b){
    return a + b
}

const restar = function(a, b){
    return a - b
}

const multiplicar = (a, b) => a * b

/* 
Manejo de Paramestros
*/

/* 1. Parametros Posicionales */

function sumar(a, b){
    return a + b
}

let resultado = sumar(10, 10) // a = 10, b = 10

/* 2. Parametros por Defecto */
function sumar(a, b = 5){
    return a + b
}

let resultado1 = sumar(10, 12) // 22
let resultado2 = sumar(10) // 15

File: funciones.py
# Funciones

"""  

def nombre_funcion(params):
    codigo
    
    
nombre_funcion = lambda params : operacion 


"""

def sumar(a, b = 15):
    return a + b

restar = lambda a, b: a - b

# 1. Parametros Posicionales

resultado = sumar(10, 10)

# 2. Parametros por Defecto
resultado2 = sumar(10) # 25

# 3. Argumentos Nombrados

resultado = sumar(b=45, a=20)

# 4. *args (Argumentos Variables)

def sumar(*numeros):
    return sum(numeros)

print(sumar(1, 2, 3, 4, 5, 6, 7))
print(sumar(1, 2, 3, 4))

# 5. **kargs (Argumentos Clave-Valor)

def mostrar_datos(**datos):
    print(f"Hola mi nombre es: {datos["nombre"]} {datos["apellido"]}")
    
mostrar_datos(nombre="Luis", apellido="Rodriguez")


def ejemplo(a, b=10, *args, **kargs):
    print(a, b)
    print(args)
    print(kargs)

print("Combinado:")
ejemplo(5, 12, 4, 6, nombre="John", apellido="Doe")
    